### PR TITLE
Add pipelinerun visualization in details page

### PR DESCRIPTION
**Pull request by karthikjeeyar**
## Fixes 

https://issues.redhat.com/browse/HACBS-444

## Description

Pipelinerun visualization graph

1. Contains horizontal scrollbar ( till we get the zoom level feature from PF package).

## Type of change

- [x] Feature

## Screen shots / Gifs for design review 

image

With scrollbars:

image

GIF:

pipelinerun-graph

## How to test or reproduce?

1. Create a component and merge the PR
2. Visit the pipelinerun details page.

## Unit Tests

```
create-utils
  ✓ should return default values if extra data is not passed (2 ms)
  ✓ should return with passed data (1 ms)

pipelinerun-graph-utils -> appendStatus
  ✓ should append Idle status if a taskrun status reason is missing
  ✓ should append Skipped status for the skipped tasks
  ✓ should append Idle status if the taskruns are missing and overall pipelinerun status is PipelineRunPending
  ✓ should append Cancelled status if the taskruns are missing and overall pipelinerun status is Cancelled (3 ms)
  ✓ should append Idle status if the status is missing in pipelinerun and taskrun
  
PipelineRunTaskRunsTab
  ✓ should not render the TaskRun table if pipelinerun is missing (18 ms)
  ✓ should render TaskRun empty state (11 ms)
  ✓ should render TaskRun table (100 ms)
    
PipelineRunDetailsTab
  ✓ should not render the pipelinerun details tab (21 ms)
  ✓ should render the pipelinerun details tab (31 ms)
  ✓ should not render the pipelinerun visualization if the status field is missing (15 ms)
  ✓ should render the pipelinerun visualization (17 ms)
      
PipelineRunVisualization
  ✓ should not render the pipelinerun graph (16 ms)
  ✓ should render the pipelinerun graph (11 ms)
  
layoutFactory
  ✓ should return null for invalid values (2 ms)
  ✓ should return layout for valid layouts options

VisualizationFactory
  ✓ should not pass controlBar to TopologyView (22 ms)
  ✓ should pass controlBar to TopologyView (7 ms)

```  
## Browser conformance: 

- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

**Comment by karthikjeeyar** [comment]
@Ranelim @MariaLeonova Please take a look

**Comment by karthikjeeyar** [comment]
cc: @christianvogt @jeff-phillips-18

File: src/hacbs/components/ApplicationDetails/tabs/overview/visualization/utils/draw-utils.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ import { Point } from '@patternfly/react-topology';
 import { DrawDesign, NODE_SEPARATION_HORIZONTAL } from '../const';
 
 type SingleDraw = (p: Point) => string;
-type DoubleDraw = (p1: Point, p2: Point, indentedStart?: number) => string;
+type DoubleDraw = (p1: Point, p2: Point, indentedStart?: number, indentedEnd?: number) => string;
```
Comment (jeff-phillips-18) [inline src/hacbs/components/ApplicationDetails/tabs/overview/visualization/utils/draw-utils.ts:5]: We should create an issue for PF topology to handle this and remove this file completely when it does.

File: src/hacbs/components/PipelineRunDetailsView/visualization/edges/PipelineRunEdge.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import { observer } from 'mobx-react';
+import { integralShapePath } from '../../../ApplicationDetails/tabs/overview/visualization/utils/draw-utils';
+import { DEFAULT_EDGE_INDENT } from '../../../topology/const';
+
+interface PipelineRunEdgeProps {
+  element: Edge;
+  className?: string;
+}
+
+const PipelineRunEdge: React.FunctionComponent<PipelineRunEdgeProps> = ({ element, className }) => {
```
Comment (jeff-phillips-18) [inline src/hacbs/components/PipelineRunDetailsView/visualization/edges/PipelineRunEdge.tsx:14]: We should update the PF topology `TaskEdge` to handle the end indent and reuse it.

File: src/hacbs/components/topology/factories/layoutFactory.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+}
+
+const DAGRE_SHARED_PROPS: dagre.GraphLabel = {
+  nodesep: NODE_SEPARATION_VERTICAL,
+  ranksep: NODE_SEPARATION_HORIZONTAL,
+  edgesep: 60,
+  ranker: 'longest-path',
+  rankdir: 'LR',
+  marginx: 20,
+  marginy: 20,
```
Comment (jeff-phillips-18) [inline src/hacbs/components/topology/factories/layoutFactory.ts:21]: These are the PF default values, so you can drop unless you specifically want to call them out.

File: src/hacbs/components/PipelineRunDetailsView/visualization/edges/PipelineRunEdge.tsx (hunk 2 above)
Comment (karthikjeeyar) [reply src/hacbs/components/PipelineRunDetailsView/visualization/edges/PipelineRunEdge.tsx:14]: Yes, I will log an issue on PF for this usecase.

**Comment by karthikjeeyar** [comment]
rebased the pr.

**Pull request merged**